src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -1771,6 +1789,46 @@ private void RegisterHomeWorkspace(HomeWorkspaceModel newWorkspace)
             };
         }
 
+        /// <summary>
+        /// Collects assemblies used by the current workspace
+        /// </summary>
+        /// <returns></returns>
+        private HashSet<AssemblyName> OnCollectingAssembliesUsed()
+        {
+            var assemblyNames = new HashSet<AssemblyName>();
+            foreach(var node in CurrentWorkspace.Nodes)
```

Comment:
this has the possibility of being quite slow... I wonder if we can turn this off during back up save? Can we look at the performance impact of this?

Reply:
I'll look into both of these ideas